integration-tests/e2e/kubetest/results_evaluator.go

```diff
@@ -66,70 +66,111 @@ func writeSummaryToFile(summary Summary) {
+		if err = junitXMLTestcasesToJSON(junitXml.Testcases); err != nil {
+			return err
+		}
+		for _, newTestcase := range junitXml.Testcases {
+			_, existsInFailureOccurences := failureOccurrences[newTestcase.Name]
+			_, existsInSucceededTestcases := succeededTestcases[newTestcase.Name]
+			if newTestcase.Skipped {
+				if !existsInFailureOccurences && !existsInSucceededTestcases {
+					skippedTestcases[newTestcase.Name] = newTestcase
 				}
+				// skipped testcases are appended later, to avoid duplicates. Therefore continue with next testcase
 				continue
 			}
-			if !testcase.Successful && !testcase.Skipped {
-				if _, ok := failureOccurrences[testcase.Name]; ok {
-					failureOccurrences[testcase.Name]++
+			testcases = append(testcases, newTestcase) // collect testcases that are either failed or succeeded
+			delete(skippedTestcases, newTestcase.Name)// if testcase was indexed as skipped from previous junit xml files, remove element
+			if newTestcase.Successful {
+				succeededTestcases[newTestcase.Name] = true
+			} else {
+				if existsInFailureOccurences {
+					failureOccurrences[newTestcase.Name]++
 				} else {
-					failureOccurrences[testcase.Name] = 1
+					failureOccurrences[newTestcase.Name] = 1
 				}
 			}
-			testcaseNameToTestcase[testcase.Name] = testcase
-			testcaseJSON, err := json.Marshal(testcase)
-			if err != nil {
-				return errors.Wrapf(err, "Couldn't marshal testsuite summary %s", testcaseJSON)
-			}
-			testcaseJSON = append([]byte("{\"index\": {\"_index\": \"e2e_testcase\", \"_type\": \"_doc\"}}\n"), testcaseJSON...)
-
-			jsonFileName := fmt.Sprintf("test-%s.json", strconv.FormatInt(time.Now().UnixNano(), 10))
-			testcaseJsonFilePath := path.Join(config.ExportPath, jsonFileName)
-			if err := ioutil.WriteFile(testcaseJsonFilePath, testcaseJSON, 0644); err != nil {
-				return errors.Wrapf(err, "Couldn't write %s to file", testcaseJsonFilePath)
-			}
 		}
 	}
-	for _, testcase := range testcaseNameToTestcase {
+	for _, testcase := range testcases {
 		mergedJunitXmlResult.Testcases = append(mergedJunitXmlResult.Testcases, testcase)
 	}
-	addFlakynessInfoToSummary(summary, &failureOccurrences)
+	for _, testcase := range skippedTestcases {
+		mergedJunitXmlResult.Testcases = append(mergedJunitXmlResult.Testcases, testcase)
+	}
+	addAdditionalInfoToSummary(summary, &failureOccurrences, &succeededTestcases)
 	if err := saveJunitXmlToFile(mergedJunitXmlResult); err != nil {
 		return err
 	}
 	return nil
 }
 
-func addFlakynessInfoToSummary(summary *Summary, failureOccurrences *map[string]int) {
+func junitXMLTestcasesToJSON(junitXMLTestcases []TestcaseResult) error {
+	for _, newTestcase := range junitXMLTestcases {
+		if !newTestcase.Skipped {
+			// write only not skipped testcases to json file
+			if err := writeTestcaseToJSONFile(newTestcase); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func unmarshalJUnitXMLResult(junitXMLPath string) (JunitXMLResult, error) {
```

Comment:
maybe rename to `unmashalJUnitFromFile` because there is already a method called `UnmarshalJunitXMLResult`. Just that the difference is clear